lib/components/artist/shows.js

```diff
@@ -7,29 +7,101 @@ const { View, Text, StyleSheet, Image, ListView } = React;
 import Headline from '../text/headline';
 import SerifText from '../text/serif'
 import Show from './show'
+import OpaqueImageView from '../opaque_image_view'
+
+var url = 'https://www.pangeareptile.com/Images/care/baby.jpg'
 
-export default class Shows extends React.Component {
 
-  updateShowData(props) {
-    // this.setState({
-    //   dataSource: this.state.dataSource.cloneWithRows(this.props.show),
-    //   loaded: true,
-    // });
+export default class Shows extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentShowsDataSource: new ListView.DataSource({
+        rowHasChanged:(row1, row2) => row1 !== row2,
+      }),
+      pastShowsDataSource: new ListView.DataSource({
+        rowHasChanged:(row1, row2) => row1 != row2,
+      }),
+      loaded: false,
+    };
   }
 
-  renderShow() {
-    return <ShowView show={this.props.show} />
+  componentDidMount() {
+    this.setState({
```

Comment:
If you try to change the state inside ```render( )```, RN will graciously tell you "Cannot update during an existing state transition (such as within `render`)", so state changes need to happen elsewhere. In Facebook's tutorial, they put this inside ```componentDidMount``` alongside network requests, so I adopted the pattern.